### PR TITLE
Don't run ntp fix from ncn-m002 --> ncn-m001

### PR DESCRIPTION
**Pull request by jacobsalmela**
## Summary and Scope

- The recent NTP fix that was introduced cannot actually run from ncn-m002 during stage 2.3 of the upgrade due to the lack of the tarball being available on ncn-m002.  To solve this, a simple check is now in place to skip ncn-m001.
- The second part involves running the script on m001 locally, at the very start of the install during `prerequisites.sh`

## Issues and Related PRs

* Resolves CASMINST-4714

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `#drax`

### Test description:

Ran the code in `ncn-rebuild-common.sh` from ncn-m002 during stage 2.3 of the upgrade:

```
root@ncn-m002 2022-05-26 18:27:59 ~ # /usr/share/doc/csm/upgrade/1.2/scripts/upgrade/ncn-upgrade-master-nodes.sh ncn-m001


 ************
 *** NOTE ***
 ************
LOG_FILE is not specified; use default location: /root/output.log

Updating SSH keys for node ncn-m001 with IP address of 10.252.1.4
====> BACKUP_SAT_LOCAL_FILES has been completed
====> BACKUP_M001_NET_CONFIG has been completed
====> PREPARE_ETCD has been completed
====> DRAIN_NODE has been completed
====> CSI_VALIDATE_BSS_NTP has been completed
====> ELIMINATE_NTP_CLOCK_SKEW ...
====> WIPE_NODE_DISK ...
```

The other portion is the same code, but is as of yet, untested in the procedure.

## Risks and Mitigations

Same as any other NTP bandage we have in place.

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: upgrade/1.2/scripts/rebuild/prerequisites.sh [hunk 1]
```diff
@@ -57,4 +57,36 @@ else
...
+state_name="ELIMINATE_NTP_CLOCK_SKEW_M001"
+state_recorded=$(is_state_recorded "${state_name}" "${target_ncn}")
+if [[ $state_recorded == "0" && $2 != "--rebuild" ]]; then
+    echo "====> ${state_name} ..."
+    {
+    if [[ "${target_ncn}" == "ncn-m001" ]]; then
+      # ensure the directory exists
+      mkdir -p /srv/cray/scripts/common/
+      # copy the ntp script and template to ncn-m001
+      cp -r "${CSM_ARTI_DIR}"/chrony/ /srv/cray/scripts/common/
```
Comment (mharding-hpe) [inline upgrade/1.2/scripts/rebuild/prerequisites.sh:69]: Should `-p` or `--preserve=mode` be used to preserve execute permission?

```suggestion
      cp -r --preserve=mode "${CSM_ARTI_DIR}"/chrony/ /srv/cray/scripts/common/
```

**Review by mharding-hpe — APPROVED** [review]
Looks good. None of my suggestions are big deals. I approve whether or not any of them are used. Don't forget to make a main branch PR

File: upgrade/1.2/scripts/rebuild/prerequisites.sh [hunk 2]
```diff
@@ -57,4 +57,36 @@ else
     echo "====> ${state_name} has been completed"
 fi
 
+state_name="ELIMINATE_NTP_CLOCK_SKEW_M001"
+state_recorded=$(is_state_recorded "${state_name}" "${target_ncn}")
+if [[ $state_recorded == "0" && $2 != "--rebuild" ]]; then
```
[the remark this answers is not in the document]
Comment (jacobsalmela) [reply upgrade/1.2/scripts/rebuild/prerequisites.sh:62]: Nope, but shellcheck complains, so I trust that.

**Pull request merged**